internal/goofys.go

```diff
@@ -1049,14 +1049,14 @@ func (fs *Goofys) MkDir(
 	parent := fs.getInodeOrDie(op.Parent)
 	fs.mu.RUnlock()
 
+	parent.mu.Lock()
+
 	// ignore op.Mode for now
 	inode, err := parent.MkDir(op.Name)
 	if err != nil {
 		return err
 	}
 
-	parent.mu.Lock()
-
```

Comment:
The folder creation is moved after acquiring the parent lock. This lock needs to be atomic between creating inode and inserting into map of inodes.